internal/gatewayapi/translator.go

```diff
@@ -1095,6 +1095,13 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 						}
 					}
 
+					if match.Method != nil {
+						irRoute.HeaderMatches = append(irRoute.HeaderMatches, &ir.StringMatch{
+							Name:  ":method",
+							Exact: StringPtr(string(*match.Method)),
```

Comment:
does the `method` type validations happen in the client side or do we need to add some checks in here ?

Reply:
there are kubebuilder validations for this (https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/httproute_types.go#L475) but yeah probably a good idea to re-validate here.

Reply:
oh sweet in that case I say skip the validations here :)